text/0000-refactor-sbom.md

```diff
@@ -0,0 +1,496 @@
+ - via Referrers Tag Schema (fallback) if Referrers API is not supported by the registry
+   - a manifest "list" published to tag `sha256-<image digest>` where each list element should contain:
+     - `artifactType` corresponding to the `config.mediaType` in the manifest
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this? Our current approach has a few drawbacks, namely:
+  * It can make application images quite large
+  * It can be hard to find SBOMs for buildpacks-built images; ecosystem tooling such as `cosign download sbom` won't work (see [PR 278](https://github.com/buildpacks/rfcs/pull/278))
+  * There is currently no clear way to associate SBOMs for build and run base images with an application image (today this is unspec'd and entirely up to the platform to manage)
+  
+- What use cases does it support?
+  * As a buildpacks user, I want **signed** SBOMs for my images in an easily consumable format.
+  * ...
+
+- What is the expected outcome?
+  * Smaller application images
+  * Integration with ecosystem tooling
+  * (Eventually) A more complete SBOM for buildpacks-built images
+
+# What it is
+[what-it-is]: #what-it-is
+
+See [How it Works](#how-it-works).
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Setup
+
+The lifecycle will ship with a new `signer` binary.
+In the past we've talked about making this binary totally separate from the lifecycle,
+but given the need to restore SBOM data and thus teach the lifecycle about attestations, there isn't much value there.
+We should make it possible to invoke the `signer` separately from other phases.
+It should not be necessary to run the `signer` within the build container (although it might be more convenient to run it that way).
+
+Buildpacks should continue to output SBOM files as before. We won't need to bother buildpack authors with this change.
+
+## Export
+
+The `exporter` will accept a new optional flag `-omit-sbom` that defaults to `false`.
+* If `-omit-sbom=false` the exporter will behave as today, and include the SBOM layer containing the `<layers>/sbom/launch/` directory in the application image.
+* If `-omit-sbom=true` the exporter will not create the SBOM layer, but `<layers>/sbom/launch/` will remain on the filesystem and be available to the `signer`.
+
+The exporter will additionally generate configuration for the `signer` irrespective of the value of `-omit-sbom`.
+
+## Sign
+
+The `signer` will accept configuration that will allow it to sign images with `cosign`. 
+An example `cosign.toml` was provided in [PR 195](https://github.com/buildpacks/rfcs/pull/195), however there were concerns about encoding sensitive data (such as passwords) in files.
+We should explore secure ways of allowing the platform to inject this data (TODO).
+It's worth noting that `cosign` as of v2 prefers keyless signing.
+
+The `signer` can both sign the image and attach attestations to it. By default, we should do both - but we may want to make this configurable.
```

Comment:
`signer` is then a registry operation?  This means that both `signer` and `exporter` now write to the registry?